### PR TITLE
Auto-select ComboBoxItem based on keyboard inputs (select-as-you-type)

### PR DESCRIPTION
**Pull request by luthfiampas**
## What does the pull request do?
Adds auto-select functionality into `ComboBox` control by capturing keyboard inputs. In WPF, we have this kind of functionality. 
Auto Select

## What is the updated/expected behavior with this PR?
It adds `IsAutoSelectEnabled` property.

## How was the solution implemented (if it's not obvious)?
- It will try to find and select item if its `Content.ToString()` starts with the search term
- It utilizes `DispatcherTimer` to reset search term on idle

## Checklist
- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by luthfiampas** [comment]
This is ready for review, thanks.

**Comment by grokys** [comment]
Great, thanks @luthfiampas - this is something I've been meaning to implement for ages. Will take a look ASAP.

File: src/Avalonia.Controls/ComboBox.cs [hunk 1]
```diff
@@ -76,6 +77,14 @@ public class ComboBox : SelectingItemsControl
         public static readonly StyledProperty<VerticalAlignment> VerticalContentAlignmentProperty =
             ContentControl.VerticalContentAlignmentProperty.AddOwner<ComboBox>();
 
+        /// <summary>
+        /// Defines the <see cref="IsAutoSelectEnabled"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsAutoSelectEnabledProperty =
```
Comment (grokys) [inline src/Avalonia.Controls/ComboBox.cs:83]: Do we need this property? In WPF and UWP as far as I can see this is just default behavior and not sure what the use-case would be for disabling it?

**Review by grokys — COMMENTED** [review]
LGTM! Thanks so much for this. Only thing stopping me from approving it is that I'm not sure we need to hide it behind a switch?

File: src/Avalonia.Controls/ComboBox.cs (hunk 1 above)
Comment (pr8x) [reply src/Avalonia.Controls/ComboBox.cs:83]: I personally like the idea of disabling it actually. Giving more control of the interactions of a control is never bad IMO.

Comment (luthfiampas) [reply src/Avalonia.Controls/ComboBox.cs:83]: The reason why I created this property; for instance, if all of ComboBox items are editable input control, I think it'll be good idea to disable this functionality. Or maybe we can set the defaut value to true?

Comment (luthfiampas) [reply src/Avalonia.Controls/ComboBox.cs:83]: @grokys The reason I created this property; for instance, if all of ComboBox items are editable input control, I think it'll be good idea to disable this functionality. Or, how about set the defaut value to true?

Comment (grokys) [reply src/Avalonia.Controls/ComboBox.cs:83]: Well if the ComboBox items are editable then those controls should set `e.Handled` on the event and it shouldn't reach the `ComboBox`. However if we feel that it'd be useful to be able to disable this behavior I'd be OK with keeping a property to enable it.

I do think though that the default should be `true` like you say, and `IsAutoSelectEnabled` perhaps isn't descriptive enough?

Looking at WPF it seems this behavior can be configured by setting [`TextSearch.TextPath`](https://wpf.2000things.com/tag/textsearch-textpath/) so looks like the feature there is called "text search". Maybe the property should be called `IsTextSearchEnabled`?

Comment (luthfiampas) [reply src/Avalonia.Controls/ComboBox.cs:83]: @grokys Okay, I agree with that :thumbsup:

**Review by grokys — APPROVED** [review]

**Comment by grokys** [comment]
LGTM! Thanks a lot @luthfiampas !

**Pull request merged**